dask/dataframe/tests/test_io.py

```diff
@@ -729,3 +729,37 @@ def test_from_pandas_with_datetime_index():
                          parse_dates=['Date'])
         ddf = dd.from_pandas(df, 2)
         eq(df, ddf)
+
+
+ar = pd.Series(range(0, 100))
+br = ar % 7
+cr = br * 3.3
+dr = br / 1.9836
+test_df = pd.concat([ar, br, cr, dr], axis=1)
+
+
+def test_encoding_gh601():
+    with tmpfile('.csv') as fn:
+        test_df.to_csv(fn, encoding='utf-16')
+
+        a = pd.read_csv(fn, encoding='utf-16')
+        d = dd.read_csv(fn, encoding='utf-16', chunkbytes=1000)
+        d.compute()
+
+
+def test_encoding_gh601_utf_16_le():
+    with tmpfile('.csv') as fn:
+        test_df.to_csv(fn, encoding='utf-16-le')
+
+        a = pd.read_csv(fn, encoding='utf-16-le')
+        d = dd.read_csv(fn, encoding='utf-16-le', chunkbytes=1000)
+        d.compute()
+
+
+def test_encoding_gh601_utf_16_be():
+    with tmpfile('.csv') as fn:
+        test_df.to_csv(fn, encoding='utf-16-be')
+
+        a = pd.read_csv(fn, encoding='utf-16-be')
+        d = dd.read_csv(fn, encoding='utf-16-be', chunkbytes=1000)
+        d.compute()
```

Comment:
Should we test for equality with `eq(d, test_df)`?

Should these tests be merged together with a for loop or pytest fixture?